src/bin/cargo/cli.rs

```diff
@@ -690,10 +690,39 @@ See '<cyan,bold>cargo help</> <cyan><<command>></>' for more information on a sp
                     help.into()
                 }))
             }).collect()
+        })))
+                .arg( Arg::new("toolchain")
+            .help("Specifies the toolchain to use for this command")
+            .value_name("TOOLCHAIN")
+            .action(ArgAction::Set)
+        .add(clap_complete::engine::ArgValueCompleter::new(|partial: &OsStr| {
+            let toolchains = get_toolchains_from_rustup();
+
+            toolchains
+                .into_iter()
+                .filter(|t| OsStr::new(t).as_encoded_bytes().starts_with(partial.as_encoded_bytes()))
```

Comment:
Is this OsStr conversion necessary? `CompletionCandidate::new` seems to accept any `Into<OsString>` type.